src/components/Body/WritePost.jsx

```diff
@@ -0,0 +1,76 @@
+import { useContext, useState } from "react";
+import { PostContext } from "../../App.jsx";
+import "./Body.css";
+
+const INITIAL_POST = {
+  title: "",
+  content: "",
+};
+
+const currentUserId = 1; //This is a placeholder
+export default function WritePost() {
+  const { posts, setPosts } = useContext(PostContext);
+  const [newPost, setNewPost] = useState({ title: "", content: "" });
+
+  const handleFormChange = (event) => {
+    const { value } = event.target;
+    setNewPost({ ...newPost, content: value });
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    // Check that newPost is not empty
+    if (newPost.content !== "" && newPost.content !== undefined) {
+      // Splitting of the first sentence to become title
+      const tempArray = newPost.content.split(".");
+      newPost.title = tempArray[0];
```

Comment:
When updating newPost before sending it to the server, you should avoid directly mutating the state. Instead, create a new object that includes all necessary properties

Reply:
Can you expand on this? I don't quite get it.